### ISSUE TITLE
Ed25519 support through the pk module

### ISSUE THREAD

**Issue by gilles-peskine-arm** [ARMmbed/mbedtls#3760]
* Add support for Ed25519 and for Ed25519ph with an empty context through `mbedtls_pk_sign` and `mbedtls_pk_verify`.
* Unit tests.

Restartable operation is out of scope.

Prerequisites:

* Ed25519 implementation: https://github.com/ARMmbed/mbedtls/issues/3757

Note: this may be done separately for Ed25519(ph) and Ed448(ph).

Foreseen extensions:

* Add support for EdDSA with a context in the pk module.
* X.509 support.
* TLS support.

**Comment by gilles-peskine-arm** [comment]
Note that the pk interface assumes that the input of the signature interface is a hash. So this task only applies to Ed25519ph. Ed25519 (where data derived from the private key is fed to the hash) is out of scope.

**Comment by sipiyou** [comment]
Is ed25519ph now fully implemented ? Could someone please post an example snippet on how to use this ?

**Comment by gilles-peskine-arm** [comment]
No, EdDSA is not implemented yet. There's a [pull request for it](https://github.com/Mbed-TLS/mbedtls/pull/5819) (just the algorithm, not the pk interface), which we currently lack the bandwidth to review.

**Comment by georgegan2023** [comment]
hi，do you have plans to develop x509 certificate parsing with ED2519 signature?

**Comment by gilles-peskine-arm** [comment]
@georgegan2023 There's currently no issue for that, so please raise one if you're interested. I expect that it'll be easy once we have EdDSA itself, but we need to track that separately since it won't be automatic (we'll at least need OIDs).

Please note that EdDSA is something we definitely want, but at the moment we can't commit to when we'll have it.
